minard/polling.py

```diff
@@ -112,13 +123,131 @@ def polling_info(data_type, run_number):
 
     return data
 
+
+def most_recent_run(all_runs, requested_run):
+    '''
+    Loops over all CMOS/Base polling runs and
+    returns the run closest (>) to the request run
+    '''
+
+    run = 0
+    for runs in all_runs:
+       # Get most recent run 
+       if requested_run == 0:
+          run = runs['run']
+          break
+       # Get most recent polling run before requested run
+       if requested_run >= runs['run']:
+          run = runs['run']
+          break
+
+    return run
+
+
+def polling_summary(run):
+    ''' 
+    Get the crate average base currents and cmos rates 
+    for the run nearest to the requested run.
+    '''
+
+    # List of polling runs
+    cmos_runs, base_runs = polling_runs()
+
+    # Runs closest to requested run
+    crun = most_recent_run(cmos_runs, run)
+    brun = most_recent_run(base_runs, run)
```

Comment:
Profiling this, it takes a few seconds to load the page, and these two lines are responsible for most of the time. I profiled it and I think it would be a lot faster to do a query like this:

```
$ time psql -U snoplus -h dbug detector -c "SELECT run FROM cmos WHERE run < 104185 LIMIT 1"
  run   
--------
 103296
(1 row)


real    0m0.014s
user    0m0.003s
sys     0m0.004s
```

which only takes a couple milliseconds versus getting all runs:

```
$ time psql -U snoplus -h dbug detector -c "SELECT DISTINCT on (run) run FROM base ORDER BY run desc" > /dev/null

real    0m0.964s
user    0m0.006s
sys     0m0.003s
```

which takes almost a full second (down from ~3.7 seconds before I added an index to try to speed it up).